code/code/misc/limits.cc

```diff
@@ -96,20 +96,22 @@ int baseHp()
 
 float classHpPerLevel(const TPerson *tp){
   float hpgain=0;
-
+  int num_classes=0;
 
   for(int i=0;i<MAX_CLASSES;++i){
     if(tp->hasClass(classInfo[i].class_num)){
-      hpgain = classInfo[i].hp_per_level;
-      break;
+      hpgain += classInfo[i].hp_per_level;
+      num_classes++;
     }
   }
 
-  if(!hpgain){
+  if(!hpgain || num_classes == 0){
     vlogf(LOG_BUG, format("No class in classHpPerLevel() for %s") %  tp->getName());
     hpgain=7.0;
   } 
 
+  hpgain = hpgain / num_classes;
```

Comment:
You're casting it back to int, thus losing precision. Instead, let's do this:
```cpp
return hpgain / num_classes;
```

Reply:
I updated it. Looks nicer this way too. Thanks!